plugins/inputs/cisco_telemetry_mdt/cisco_telemetry_mdt.go

```diff
@@ -66,12 +103,50 @@ func (c *CiscoTelemetryMDT) Start(acc telegraf.Accumulator) error {
 		return err
 	}
 
+	c.propMap = make(map[string]func(field *telemetry.TelemetryField, value interface{}) interface{}, 100)
+	c.propMap["test"] = nxosValueXformUint64Toint64
+	c.propMap["asn"] = nxosValueXformUint64ToString            //uint64 to string.
+	c.propMap["subscriptionId"] = nxosValueXformUint64ToString //uint64 to string.
+	c.propMap["operState"] = nxosValueXformUint64ToString      //uint64 to string.
+
 	// Invert aliases list
 	c.warned = make(map[string]struct{})
 	c.aliases = make(map[string]string, len(c.Aliases))
 	for alias, path := range c.Aliases {
 		c.aliases[path] = alias
 	}
+	c.initDb()
+
+	c.dmes = make(map[string]string, len(c.Dmes))
+	for dme, path := range c.Dmes {
+		c.dmes[path] = dme
+		if path == "uint64 to int" {
+			c.propMap[dme] = nxosValueXformUint64Toint64
+		} else if path == "uint64 to string" {
+			c.propMap[dme] = nxosValueXformUint64ToString
+		} else if path == "string to float64" {
+			c.propMap[dme] = nxosValueXformStringTofloat
+		} else if path == "string to uint64" {
+			c.propMap[dme] = nxosValueXformStringToUint64
+		} else if path == "string to int64" {
+			c.propMap[dme] = nxosValueXformStringToInt64
+		} else if path == "auto-float-xfrom" {
+			c.propMap[dme] = nxosValueAutoXformFloatProp
+		} else if dme[0:6] == "dnpath" { //path based property map
+			js := []byte(path)
+			var jsStruct NxPayloadXfromStructure
+
+			err := json.Unmarshal(js, &jsStruct)
+			if err == nil {
+				//Build 2 level Hash nxpathMap Key = jsStruct.Name, Value = map of jsStruct.Prop
+				//It will override the default of code if same path is provided in configuration.
+				c.nxpathMap[jsStruct.Name] = make(map[string]string, len(jsStruct.Prop))
+				for _, prop := range jsStruct.Prop {
+					c.nxpathMap[jsStruct.Name][prop.Key] = prop.Value
+				}
+			}
+		}
```

Comment:
We should make this a switch statement
```suggestion
		switch path {
		case "uint64 to int":
			c.propMap[dme] = nxosValueXformUint64Toint64
		case "uint64 to string":
			c.propMap[dme] = nxosValueXformUint64ToString
		case "string to float64":
			c.propMap[dme] = nxosValueXformStringTofloat
		case "string to uint64":
			c.propMap[dme] = nxosValueXformStringToUint64
		case "string to int64":
			c.propMap[dme] = nxosValueXformStringToInt64
		case "auto-float-xfrom"
			c.propMap[dme] = nxosValueAutoXformFloatProp
		default:
			if ! strings.HasPrefix(dme, "dnpath") { // not path based property map
				continue
			}
			
			var jsStruct NxPayloadXfromStructure
			if err := json.Unmarshal([]byte(path), &jsStruct); if err != nil {
				continue
			}
				
			// Build 2 level Hash nxpathMap Key = jsStruct.Name, Value = map of jsStruct.Prop
			// It will override the default of code if same path is provided in configuration.
			c.nxpathMap[jsStruct.Name] = make(map[string]string, len(jsStruct.Prop))
			for _, prop := range jsStruct.Prop {
				c.nxpathMap[jsStruct.Name][prop.Key] = prop.Value
			}
		}
```